plugins/permission-react/src/hooks/usePermission.ts

```diff
@@ -61,6 +62,39 @@ export function usePermission(
       },
 ): AsyncPermissionResult {
   const permissionApi = useApi(permissionApiRef);
+  const configApi = useApi(configApiRef);
+
+  const enableDataloaderRequests =
+    configApi.getOptionalBoolean(
+      'permission.EXPERIMENTAL_enableDataloaderRequests',
+    ) ?? false;
+
+  if (enableDataloaderRequests) {
+    // Disable this for now until dataloader is either de-factor or removed.
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const { value, error } = useAsync(async () => {
+      // We could make the resourceRef parameter required to avoid this check, but
+      // it would make using this hook difficult in situations where the entity
+      // must be asynchronously loaded, so instead we short-circuit to a deny when
+      // no resourceRef is supplied, on the assumption that the resourceRef is
+      // still loading outside the hook.
+      if (isResourcePermission(input.permission) && !input.resourceRef) {
+        return AuthorizeResult.DENY;
+      }
+
+      const { result } = await permissionApi.authorize(input);
+      return result;
+    }, [permissionApi, input]);
```

Comment:
since `input` changes on every re-render, this will cause `permissionApi.authorize()` to be invoked multiple times with the same permissions. Will this send more requests?